### PR TITLE
[X86][MC] Fix wrong encoding of promoted BMI instructions due to missing NoCD8

### PR DESCRIPTION
**Pull request by XinWang10**
Address review comments in #76709

Add `NoCD8` to class `ITy`, and rewrite the promoted instructions with `ITy` to avoid unexpected incorrect encoding about `NoCD8`.

### CODE REVIEW

**Comment by topperc** [comment]
It's good idea to mention what instruction has the wrong encoding in the patch title or description.

**Comment by KanRobert** [comment]
@XinWang10 I believe you will rewrite other promoted instructions with `ITy`, right?

**Comment by XinWang10** [comment]
> @XinWang10 I believe you will rewrite other promoted instructions with `ITy`, right?
I just find this existing error now, if others needed to be rewrited, would do it in another NFC patch.

**Review by KanRobert — APPROVED** [review]
LGTM

File: llvm/lib/Target/X86/X86InstrMisc.td [hunk 1]
```diff
@@ -1347,38 +1347,40 @@ let Predicates = [HasBMI2, NoTBM] in {
                              (MOV8ri (CountTrailingOnes imm:$mask)), sub_8bit))>;
 }
 
-multiclass bmi_pdep_pext<string mnemonic, RegisterClass RC,
+multiclass bmi_pdep_pext<string mnemonic, X86TypeInfo t,
                          X86MemOperand x86memop, SDPatternOperator OpNode,
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrMisc.td:1351]: x86memop can be removed.

Comment (KanRobert) [reply llvm/lib/Target/X86/X86InstrMisc.td]: @XinWang10 You missed `ld_frag`.

File: llvm/lib/Target/X86/X86InstrMisc.td [hunk 2]
```diff
@@ -1375,39 +1375,32 @@ let Predicates = [HasBMI2, NoTBM, HasEGPR] in {
...
-  defm PDEP64 : bmi_pdep_pext<"pdep{q}", GR64, i64mem,
-                               X86pdep, loadi64, "_EVEX">, T8, XD, REX_W, EVEX;
-  defm PEXT32 : bmi_pdep_pext<"pext{l}", GR32, i32mem,
-                               X86pext, loadi32, "_EVEX">, T8, XS, EVEX;
-  defm PEXT64 : bmi_pdep_pext<"pext{q}", GR64, i64mem,
-                               X86pext, loadi64, "_EVEX">, T8, XS, REX_W, EVEX;
+  defm PDEP32 : bmi_pdep_pext<"pdep", Xi32, X86pdep, "_EVEX">, T8, XD, EVEX;
+  defm PDEP64 : bmi_pdep_pext<"pdep", Xi64, X86pdep, "_EVEX">, T8, XD, REX_W, EVEX;
+  defm PEXT32 : bmi_pdep_pext<"pext", Xi32, X86pext, "_EVEX">, T8, XS, EVEX;
+  defm PEXT64 : bmi_pdep_pext<"pext", Xi64, X86pext, "_EVEX">, T8, XS, REX_W, EVEX;
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrMisc.td:1403]: Move T8, XD, XS into the class

Comment (XinWang10) [reply llvm/lib/Target/X86/X86InstrMisc.td:1403]: Maybe just T8?

Comment (KanRobert) [reply llvm/lib/Target/X86/X86InstrMisc.td:1403]: Ah, Yes. My mistake

File: llvm/lib/Target/X86/X86InstrMisc.td [hunk 3]
```diff
@@ -1375,39 +1375,32 @@ let Predicates = [HasBMI2, NoTBM, HasEGPR] in {
...
-                         PatFrag ld_frag, string Suffix = ""> {
-  def rr#Suffix : I<0xF5, MRMSrcReg, (outs RC:$dst), (ins RC:$src1, RC:$src2),
-                    !strconcat(mnemonic, "\t{$src2, $src1, $dst|$dst, $src1, $src2}"),
-                    [(set RC:$dst, (OpNode RC:$src1, RC:$src2))]>,
-                  NoCD8, VVVV, Sched<[WriteALU]>;
-  def rm#Suffix : I<0xF5, MRMSrcMem, (outs RC:$dst), (ins RC:$src1, x86memop:$src2),
-                    !strconcat(mnemonic, "\t{$src2, $src1, $dst|$dst, $src1, $src2}"),
-                    [(set RC:$dst, (OpNode RC:$src1, (ld_frag addr:$src2)))]>,
-                  NoCD8, VVVV, Sched<[WriteALU.Folded, WriteALU.ReadAfterFold]>;
+multiclass bmi_pdep_pext<string mnemonic, X86TypeInfo t, SDPatternOperator OpNode,
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrMisc.td:1378]: OpNode -> node

bmi_pdep_pext -> PdepPext

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**